src/libstore/globals.hh

```diff
@@ -629,6 +629,11 @@ public:
 
           Files fetched via `NIX_PATH`, `fetchGit`, `fetchMercurial`,
           `fetchTarball`, and `fetchurl` respect this TTL.
+
+          This option also affects how often the global flake registry is
+          updated.  For example, a flake URI like `nixpkgs#hello` will trigger
+          a new download of Nixpkgs if the last download is older than the
+          `tarball-ttl`.
```

Comment:
That's two separate things: checking for a new registry version, and checking whether `github:NixOS/nixpkgs` has a new version. But both are affected by the TTL.

Reply:
I didn't catch that confusion. I think this might happen more often as people will think of the flake registry as a typical package registry, whereas it's just a name registry.

Reply:
It's not clear to me what change is being suggested here

Reply:
The global flake registry at github.com/NixOS/flake-registry is a downloaded resource just like anything else that's fetched (but not a fixed output derivation). It's a feature that's unrelated to the caching of such downloaded resources, except for the fact that the registry itself is cached. In other words, the registry is not the mechanism that implements the caching.

As such, the second sentence is not an example of the first, despite both being true.

OT: Should we even call it the "global" registry? I think it's too vague, as some tools, notably git, consider the user configuration (of all things) to be "global".

Reply:
The concrete implication is that the tarball TTL doesn't just affect `nix run nixpkgs#hello`, but also `nix run github:NixOS/nixpkgs#hello` (which doesn't use the registry at all).

The registry rarely changes, and certainly what `nixpkgs` refers to doesn't change. So the effect of the TTL on registry caching is usually irrelevant. The new download of Nixpkgs is nt the result of the registry being updated, but the `github:NixOS/nixpkgs` flake being updated.

Reply:
Maybe we should present this as a list of all things affected by the TTL, e.g.

* Files downloaded by `fetchTarball`, etc.
* The registry.
* What revision Git/GitHub"refs" refer to.
* ...